### ISSUE TITLE
Showing where the error is with the expected actions assertion

### ISSUE THREAD

**Issue by subblue** [reduxjs/redux-mock-store#7]
Firstly this library is proving to be very useful when testing my app, so many thanks!

One thing I have found is that if there is a mismatch with the expected actions in my test and the actual code then it can be tricky to find exactly where the problem is - typically the mocha test just times out because the done callback hasn't been triggered.

Is there a way to log the difference between expected and actual actions called?

**Comment by EvHaus** [comment]
Definitely agree with this. It's very hard to know what went wrong and it's not obvious where I can place a console log to see what actually occurred. Would be great if there was some way to see what actions were fired vs what I was expecting to fire.

Here is a specific example. In my test I have:

```
const expectedActions = [
    {type: types.LOGOUT, message: 'test'}
];

const store = mockStore({AuthActions: []}, expectedActions, done);
```

Which was timing out. The issue is that my action actually fires with an `error` object key and not a `message` object key, and it was very difficult to track this down. The fix was:

```
const expectedActions = [
    {type: types.LOGOUT, message: 'test'}
];

const store = mockStore({AuthActions: []}, expectedActions, done);
```

If there was some way to know what actions were fired inside the mockStore, it would have been very easy to see what's wrong as you could visually see the expected vs the actual result.

**Comment by arnaudbenard** [comment]
Which version are you both running?

**Comment by subblue** [comment]
v0.0.4

**Comment by arnaudbenard** [comment]
Can you try v0.0.3, I think that v0.0.4 introduced that bug.

**Comment by chibicode** [comment]
@arnaudbenard need help on this?

**Comment by subblue** [comment]
Interesting, switching back to v0.0.3 I do now see the diff with the expected actions as hoped. However with 0.0.3, another test that involves actions triggered after an error has been caught in a promise (part of an invalid login test) is failing where it was passing with 0.0.4

**Comment by subblue** [comment]
The tiny tweak in my pull request fixes the issues I was having when testing my app. However, I'm not sure how to properly test the test in the redux-mock-store tests!

**Comment by acrognale** [comment]
+1, I am experiencing the same issue and can confirm that switching back to 0.0.3.

I had spent a while trying to figure out if one of my own tests passing was related to my usage of nock or fetch, but this is indeed now correctly failing.

In my case, with v0.0.4, running mocha with  `NODE_ENV=production mocha ./src/tests/foo.jsx -G -R spec --ui bdd --compilers js:babel-core/register --recursive -d` simply caused mocha to exit, without saying if the test passed or failed.

**Comment by arnaudbenard** [comment]
Fixed in `v0.0.6`
